follower/execution/follower.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package execution
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/badger/operation"
+
+	"github.com/optakt/flow-dps/models/dps"
+)
+
+type blockReader interface {
+	Read(blockID flow.Identifier) ([]byte, error)
+}
+
+type unmarshaler interface {
+	Unmarshal(compressed []byte, value interface{}) error
+}
+
+type Follower struct {
+	log zerolog.Logger
+
+	blocks blockReader
+	codec  unmarshaler
+	db     *badger.DB
+
+	block BlockData
+}
+
+func New(log zerolog.Logger, blocks blockReader, codec unmarshaler, db *badger.DB) *Follower {
+	f := Follower{
+		log: log,
+
+		blocks: blocks,
+		codec:  codec,
+		db:     db,
+	}
+
+	return &f
+}
+
+func (f *Follower) Height() uint64 {
+	if f.block.Block != nil {
+		return f.block.Block.Header.Height
+	}
+	return math.MaxUint64
```

Comment:
This is rather ugly. I would rather have a separate field for the height that we initialize with `math.MaxUint64`. `nil` checks are evil except for errors.